File: doctolib-frontend/src/services/user.service.ts
import { api } from "./api"
import type { User, UserCreateDTO, UserUpdateDTO } from "../types/user"

export const userService = {
  getMe: async (): Promise<User> => {
    const response = await api.get<User>("/auth/me")
    return response.data
  },

  getById: async (id: string): Promise<User> => {
    const response = await api.get<User>(`/users/${id}`)
    return response.data
},

  create: async (data: UserCreateDTO): Promise<User> => {
    const response = await api.post<User>("/users", data)
    return response.data
  },

  update: async (id: string, data: UserUpdateDTO): Promise<User> => {
    const response = await api.put<User>(`/users/${id}`, data)
    return response.data
  },

  updateSpecialty: async (specialty: string): Promise<User> => {
    const response = await api.put<User>("/users/me/specialty", { specialty })
    return response.data
  },

  getAllPractitioners: async (): Promise<User[]> => {
    const response = await api.get<User[]>("/users/practitioners")
    return response.data
  }
}

File: doctolib_mobile/lib/services/user_service.dart
import 'package:dio/dio.dart';
import '../api/dio_client.dart';

class UserService {
  Future<Map<String, dynamic>> getMe() async {
    final dio = await DioClient.withToken();
    final response = await dio.get("/auth/me");
    return response.data;
  }

  Future<Map<String, dynamic>> getById(String id) async {
    final dio = await DioClient.withToken();
    final response = await dio.get("/users/$id");
    return response.data;
  }

  Future<List<dynamic>> getAllPractitioners() async {
    final dio = await DioClient.withToken();
    final response = await dio.get("/users/practitioners");
    return response.data;
  }

  Future<Map<String, dynamic>> create(Map<String, dynamic> data) async {
    final dio = DioClient.dio;
    final response = await dio.post("/users", data: data);
    return response.data;
  }

  Future<Map<String, dynamic>> update(
    String id,
    Map<String, dynamic> data,
  ) async {
    final dio = await DioClient.withToken();
    final response = await dio.put("/users/$id", data: data);
    return response.data;
  }

  Future<Map<String, dynamic>> updateSpecialty(String specialty) async {
    final dio = await DioClient.withToken();
    final response = await dio.put(
      "/users/me/specialty",
      data: {"specialty": specialty},
    );
    return response.data;
  }
}
